storage/storage_client.go

```diff
@@ -101,7 +102,7 @@ func (c *StorageClient) getUnqualifiedName(name string) string {
 	}
 
 	nameParts := strings.Split(name, "/")
-	return strings.Join(nameParts[2:], "/")
+	return strings.Join(nameParts[len(nameParts)-1:], "/")
```

Comment:
I think this is a little better though. Since we'll have `/account/container/` and `/account/container/object`.